vega_sim/scenario/ideal_market_maker/agents.py

```diff
@@ -572,3 +589,277 @@ def initialise(self, vega: VegaServiceNull):
+            current_position = 0
+            self.bid_depth, self.ask_depth = self.OptimalStrategy(current_position)
+            self.vega.submit_simple_liquidity(
+                wallet_name=self.wallet_name,
+                market_id=self.market_id,
+                commitment_amount=self.commitamoumt,
+                fee=self.lp_fee,
+                reference_buy="PEGGED_REFERENCE_MID",
+                reference_sell="PEGGED_REFERENCE_MID",
+                delta_buy=self.bid_depth,
+                delta_sell=self.ask_depth,
+                is_amendment=False,
+            )
+            self.current_step += 1
+            return
+
+        position = self.vega.positions_by_market(
+            wallet_name=self.wallet_name, market_id=self.market_id
+        )
+
+        current_position = int(position[0].open_volume) if position else 0
+        self.bid_depth, self.ask_depth = self.OptimalStrategy(current_position)
+
+        self.vega.submit_simple_liquidity(
+            wallet_name=self.wallet_name,
+            market_id=self.market_id,
+            commitment_amount=self.commitamoumt,
+            fee=self.lp_fee,
+            reference_buy="PEGGED_REFERENCE_MID",
+            reference_sell="PEGGED_REFERENCE_MID",
+            delta_buy=self.bid_depth,
+            delta_sell=self.ask_depth,
+            is_amendment=True,
+        )
+
+        self.current_step += 1
+
+
+class InformedTrader(StateAgentWithWallet):
+    def __init__(
+        self,
+        wallet_name: str,
+        wallet_pass: str,
+        price_process: List[float],
+        market_name: str = None,
+        asset_name: str = None,
+        initial_asset_mint: float = 1e8,
+        tag: str = "",
+    ):
+        super().__init__(wallet_name + str(tag), wallet_pass)
+        self.initial_asset_mint = initial_asset_mint
+        self.price_process = price_process
+        self.current_step = 0
+        self.sim_length = len(price_process)
+        self.tag = tag
+        self.market_name = f"ETH:USD_{self.tag}" if market_name is None else market_name
+        self.asset_name = f"tDAI_{self.tag}" if asset_name is None else asset_name
+
+    def initialise(self, vega: VegaServiceNull):
+        # Initialise wallet
+        super().initialise(vega=vega)
+
+        # Get market id
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+
+        # Get asset id
+        tDAI_id = self.vega.find_asset_id(symbol=self.asset_name)
+        # Top up asset
+        self.vega.mint(
+            self.wallet_name,
+            asset=tDAI_id,
+            amount=self.initial_asset_mint,
+        )
+        self.vega.wait_for_total_catchup()
+
+    def step(self, vega_state: VegaState):
+        position = self.vega.positions_by_market(
+            wallet_name=self.wallet_name, market_id=self.market_id
+        )
```

Comment:
It may not be in there currently, and not for this PR, but we should add all traders' positions to the vega_state so it doesn't have to be called individually